### PR TITLE
[hotfix] Update OSF harvester to honor period given to retract public registrations

### PR DESCRIPTION
**Pull request by erinspace**
Harvests will now automatically happen for a 2 day period starting 4 days in the past, to honor the 48 hours promised to contributors to public projects to make the registration private.

### CODE REVIEW

**Comment by fabianvf** [comment]
./scrapi/harvesters/osf.py:19:1: F401 'settings' imported but unused

File: scrapi/harvesters/osf.py [hunk 1]
```diff
@@ -96,9 +95,10 @@ def schema(self):
         }
 
     def harvest(self, start_date=None, end_date=None):
-
-        start_date = start_date or date.today() - timedelta(settings.DAYS_BACK)
```
Comment (fabianvf) [inline scrapi/harvesters/osf.py]: This meas that we can't harvest back a configurable number of days for the OSF.

File: scrapi/harvesters/osf.py [hunk 2]
```diff
@@ -96,9 +95,10 @@ def schema(self):
...
-        end_date = end_date or date.today()
+        # Always harvest a 2 day period starting 2 days back to honor time given
```
Comment (fabianvf) [inline scrapi/harvesters/osf.py]: Why do we always harvest a 2 day period?

File: scrapi/harvesters/osf.py (hunk 1 above)
Comment (erinspace) [inline scrapi/harvesters/osf.py]: We can if we need to! If start date is passed in that will still be used, this just defaults to 4 days back

File: scrapi/harvesters/osf.py (hunk 2 above)
Comment (erinspace) [inline scrapi/harvesters/osf.py]: I just added that in as that seemed to be the standard in other harvesters - 2 days just to make sure we cover everything in the gaps between harvests

**Pull request merged**